generators/server/index.js

```diff
@@ -558,4 +579,54 @@ module.exports = class JHipsterServerGenerator extends BaseBlueprintGenerator {
     if (useBlueprints) return;
     return this._end();
   }
+
+  _configureServer(config = this.jhipsterConfig) {
+    // JWT authentication is mandatory with Eureka, so the JHipster Registry
+    // can control the applications
+    if (config.serviceDiscoveryType === EUREKA && config.authenticationType !== OAUTH2) {
+      config.authenticationType = JWT;
+    }
+
+    // Generate JWT secret key if key does not already exist in config
+    if ((config.authenticationType === JWT || config.applicationType === MICROSERVICE) && config.jwtSecretKey === undefined) {
+      config.jwtSecretKey = getBase64Secret.call(this, null, 64);
+    }
+    // Generate remember me key if key does not already exist in config
+    if (config.authenticationType === SESSION && !config.rememberMeKey) {
+      config.rememberMeKey = getRandomHex();
+    }
+
+    if (config.authenticationType === OAUTH2) {
+      config.skipUserManagement = true;
+    }
+
+    if (config.enableHibernateCache && [NO_CACHE, MEMCACHED].includes(config.cacheProvider)) {
+      this.info(`Disabling hibernate cache for cache provider ${config.cacheProvider}`);
+      config.enableHibernateCache = false;
+    }
+
+    if (!config.databaseType && config.prodDatabaseType) {
+      config.databaseType = this.getDBTypeFromDBValue(config.prodDatabaseType);
+    }
+    if (!config.devDatabaseType && config.prodDatabaseType) {
+      config.devDatabaseType = config.prodDatabaseType;
+    }
+
+    // force variables unused by microservice applications
+    if (config.applicationType === MICROSERVICE) {
+      config.websocket = NO_WEBSOCKET;
+    }
+
+    const databaseType = config.databaseType;
+    if (databaseType === NO_DATABASE) {
+      config.devDatabaseType = NO_DATABASE;
```

Comment:
This one makes no sense

Reply:
Required to fix https://github.com/jhipster/generator-jhipster/issues/16103.